rotkehlchen/exchanges/coinbase.py

```diff
@@ -160,21 +161,117 @@ def __init__(
             secret=secret,
             database=database,
         )
+        self.is_legacy_api_key=self.determine_key_type(api_key)
```

Comment:
Do not call this function here. At initialization of all exchanges `validate_api_key` is always called so any problems with the api key will be caught there. While here the exception will go unhandled.

Reply:
I initially removed the comments when I was updating the code to support only the new key format, and I removed all the legacy key code at that time. When the requirement changed to support both key formats, I reinstated the logic for both but inadvertently didn't include the original comments. I believe I should now have added all the relevant comments back into the code.